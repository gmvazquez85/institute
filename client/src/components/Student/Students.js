import React, { Component } from 'react';

class Students extends Component {
  render() {
    let { students } = this.props;

    students = students ? students : [];

    return (
      <div>
        <ul>
          {students.length <= 0
            ? 'NO DB ENTRIES YET'
            : students.map(dat => (
                <li style={{ padding: '10px' }} key={dat.id}>
                  <span style={{ color: 'gray' }}> id: </span> {dat.id} <br />
                  <span style={{ color: 'gray' }}> Name: </span> {dat.name} <br />
                  {/*< button onClick={() => this.edit(dat.id)}>EDIT</button> <br />*/}
                  {/*< button onClick={() => this.deleteFromDB(dat.id)}>DELETE</button> <br />*/}
                </li>
              ))}
        </ul>
      </div>
    );
  }
}

export default Students;
