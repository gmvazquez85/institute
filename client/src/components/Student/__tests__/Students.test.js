import React from 'react';
import { shallow } from 'enzyme';
import Students from '../Students';

describe('Students', () => {
  it('should render correctly without students', () => {
    const component = shallow(<Students />);
  
    expect(component).toMatchSnapshot();
  });
  
  it('should render correctly with students', () => {

  	const st = [{id:1, name: "gabriel"},{id:2, name:"matias"}];

    const component = shallow(<Students students={st}/>);
  
    expect(component).toMatchSnapshot();
  });
});