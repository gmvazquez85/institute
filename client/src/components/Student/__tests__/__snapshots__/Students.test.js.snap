// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Students should render correctly with students 1`] = `
<div>
  <ul>
    <li
      key="1"
      style={
        Object {
          "padding": "10px",
        }
      }
    >
      <span
        style={
          Object {
            "color": "gray",
          }
        }
      >
         id: 
      </span>
       
      1
       
      <br />
      <span
        style={
          Object {
            "color": "gray",
          }
        }
      >
         Name: 
      </span>
       
      gabriel
       
      <br />
    </li>
    <li
      key="2"
      style={
        Object {
          "padding": "10px",
        }
      }
    >
      <span
        style={
          Object {
            "color": "gray",
          }
        }
      >
         id: 
      </span>
       
      2
       
      <br />
      <span
        style={
          Object {
            "color": "gray",
          }
        }
      >
         Name: 
      </span>
       
      matias
       
      <br />
    </li>
  </ul>
</div>
`;

exports[`Students should render correctly without students 1`] = `
<div>
  <ul>
    NO DB ENTRIES YET
  </ul>
</div>
`;
